open-craft/xblock-eoc-journal
# [MCKIN-8436] Port eoc-journal block to use the completion v1 API

This removes fetching user progress from [api-integration](https://github.com/edx-solutions/api-integration) and calculating it manually in favor of retrieving calculated percentage progress from [Completion API](https://github.com/open-craft/openedx-completion-aggregator).

The redundant code was removed and the existing API client were simplified by inheriting from a `BaseApiClient`.

**Testing instructions**:
- create freeform questions from [problem-builder](https://github.com/open-craft/problem-builder),
- create end of course journal block,
- check if data in `Progress` of the journal block in the course is correct.
